presto-druid/src/main/java/com/facebook/presto/druid/segment/SmooshedColumnSource.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.druid.segment;
+
+import com.facebook.presto.spi.PrestoException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.druid.common.utils.SerializerUtils;
+import org.apache.druid.jackson.DefaultObjectMapper;
```

Comment:
Can some of this Druid specific stuff be pushed into its own repo like we did for pinot: https://github.com/prestodb/presto-pinot-driver. For example, this repo abstracts all of the pinot specific intricacies like talking to ZK, merging the server results and what not. I feel that there is some of that going in Druid land as well: you stitch results from hdfs and druid proper.

Reply:
good idea. We did not separate them out yet. Currently, Druid connector only use Druid for metadata. it is scanning Druid segments files directly from HDFS.